goldmane/pkg/storage/add_flow_bench_test.go

```diff
@@ -0,0 +1,125 @@
+package storage_test
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/projectcalico/calico/goldmane/pkg/storage"
+	"github.com/projectcalico/calico/goldmane/pkg/types"
+	"github.com/projectcalico/calico/goldmane/proto"
+	"unique"
+)
+
+var keys = []string{"app", "env", "tier", "version", "team", "region", "zone", "instance", "project", "role"}
+var values = []string{"frontend", "prod", "backend", "v1", "infra", "us-west", "us-west-2a", "i-1234", "goldmane", "web"}
+
+func generateLabelMap(prefix string, count int) map[string]string {
+	labelMap := make(map[string]string, count)
+	for i := 0; i < count && i < len(keys); i++ {
+		labelMap[fmt.Sprintf("%s-%s", prefix, keys[i])] = values[i]
+	}
+
+	return labelMap
+}
+
+func generateRandomLabelMap(r *rand.Rand, prefix string, count int) map[string]string {
+	r.Shuffle(len(keys), func(i, j int) { keys[i], keys[j] = keys[j], keys[i] })
+	r.Shuffle(len(values), func(i, j int) { values[i], values[j] = values[j], values[i] })
+
+	return generateLabelMap(prefix, count)
+}
+
+func encodeLabels(m map[string]string) unique.Handle[string] {
+	var parts []string
+	for k, v := range m {
+		parts = append(parts, fmt.Sprintf("%s=%s", k, v))
+	}
+	return unique.Make(strings.Join(parts, ","))
+}
+
+type labelPair struct {
+	src unique.Handle[string]
+	dst unique.Handle[string]
+}
+
+func BenchmarkDiachronicFlow_AddFlow(b *testing.B) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	labelPairs := []labelPair{
+		{
+			encodeLabels(generateLabelMap("src", 10)),
+			encodeLabels(generateLabelMap("dst", 10)),
+		},
+		{
+			encodeLabels(generateRandomLabelMap(r, "src", r.Intn(9)+2)),
+			encodeLabels(generateRandomLabelMap(r, "dst", r.Intn(9)+2)),
+		},
+	}
+
+	b.ResetTimer()
+	runAddFlowBenchmark(b, labelPairs)
+}
+
+func runAddFlowBenchmark(b *testing.B, labelPairs []labelPair) {
```

Comment:
Is there a reason we need to split this part out into a separate function?

Reply:
fixed